src/metadata-parser.h

```diff
@@ -222,6 +222,93 @@ namespace librealsense
         };
     };
 
+
+    class actual_fps_calculator
+    {
+    public:
+        rs2_metadata_type get_fps(const frame & frm)
+        {
+            std::lock_guard<std::mutex> lock();
+
+            auto frame_number = frm.get_frame_number();
+            if (frame_number > _last_frame_number)
```

Comment:
actual_fps_calculator::get_fps is called when there is no metadata it can be called couple of time on the same frame and also can be called on frames out of order so if the frame number is smaller its not necessary said that its, overflow.
in case of overflow, it will affect only on one frame and in this case it will return the last cashed fps.